pkg/vectorize/abs/abs.go

```diff
@@ -82,6 +88,9 @@ func AbsInt8(xs, rs []int8) []int8 {
 
 func absInt8Pure(xs, rs []int8) []int8 {
```

Comment:
Is it suitable to add bound check here ? I'm afraid it would affect the performance due to breaking the vectorized execution of cpu instruction. ClickHouse does not check the bound for the functions, neither does for Starrocks.

Reply:
Must check.    Correctness first.   I am not convinced you gain much performance anyway, modern CPU should do a pretty good job here.   If we are worried about panic, we can set a flag in the loop and check panic after the loop, but I am not sure it is a problem that worth it.

Reply:
this is the original absInt8Pure code;
```go
func absInt8Pure(xs, rs []int8) []int8 {
	for i := range xs {
		if xs[i] < 0 {
			rs[i] = -xs[i]
		} else {
			rs[i] = xs[i]
		}
	}
	return rs
}
```
if we handle bound checking like this:
```go
func absInt8Pure(xs, rs []int8) []int8 {
	for i := range xs {
		if xs[i] < 0 {
                        if xs[I] == int8(-128) {
                            rs[i] = SOME_SPECIFIC_VALUE
                        } else {
                            rs[i] = -xs[i]
                        }
		} else {
			rs[i] = xs[i]
		}
	}
	return rs
}
```

I think we would suffer at least a two fold cost penalty in the xs[I] < 0 branch...

most importantly, since we may vectorize this function later, we must keep our pure go version implementation consistent with the cpu intrinsic's behavior, and from my little experiment, it seems like cpu wouldn't do this bound checking either.
```c
 #include <immintrin.h>
  #include <stdio.h>

  union U256i8 {
          __m256i v;
          int8_t a[32];
  };

  void print(__m256i v) {
          const union U256i8 u = {v};

          for (int i = 0; i < 32; i++) {
                  printf("%i\n", u.a[i]);
          }
  }

  void test(int8_t value) {
          __m256i v1 = _mm256_set1_epi8(value);
          __m256i a = _mm256_abs_epi8(v1);
          printf("abs %d:\n", value);
          print(a);
  }

  int main() {
          test(-23);
          test(-128);
  }
```
output:
abs -128:
-128
-128
-128
-128
-128
-128
-128
.......


anyways, this is more like product design issue...

Reply:
in this specific scenario, perhaps we just need to change our return type to uint8